lib/generators/accessly/install/templates/db/migrate/create_permitted_actions.rb

```diff
@@ -0,0 +1,13 @@
+class CreatePermittedActions < ActiveRecord::Migration<%= migration_version %>
+  def change
+    create_table :accessly_permitted_actions, force: true, id: :uuid do |t|
+      t.column :segment_id, :integer, default: -1
+      t.column :action, :integer, null: false
+      t.column :actor_id, :integer, null: false
+      t.column :actor_type, :string, null: false
+      t.column :object_type, :string, null: false
+    end
+
+    add_index(:accessly_permitted_actions, [:segment_id, :action, :actor_id, :actor_type, :object_type], unique: true, name: "acessly_pa_uniq_table_idx")
```

Comment:
@ehourigan I would argue that this index should be:

```rb
[:segment_id, :actor_id, :actor_type, :object_type, :action]
```

Because it seems like we're more likely to get a list of all the actions a user can do or to get a list of all the actions a user can do in a given namespace (object_type) than to get a list of all the actors who can do a particular action.